dialoguekit/core/annotated_utterance.py

```diff
@@ -40,6 +43,25 @@ def from_utterance(cls, utterance: Utterance):
         args = asdict(utterance)
         return cls(**args)
 
+    def add_dialogue_acts(self, dialogue_acts: List[DialogueAct]) -> None:
+        """Adds dialogue acts to the utterance.
+
+        Args:
+            dialogue_acts: List of dialogue acts.
+        """
+        self.dialogue_acts.extend(dialogue_acts)
+
+    def get_intents(self) -> List[Intent]:
+        """Returns utterance's intents."""
+        return [da.intent for da in self.dialogue_acts]
+
+    def get_dialogue_act_annotations(self) -> Set[Annotation]:
```

Comment:
Is there a specific reason for this extra conversion to set? It'd simplify things if we returned a list. In most cases, I'd expect that the return value would simply be iterated over. The caller method can convert to a set, if needed.

Reply:
The idea was to remove duplicated slot value pairs. I applied to suggested modification.